lua/scratch/utils.lua

```diff
@@ -102,4 +102,118 @@ function M.getSelectedText()
   return lines
 end
 
-return M
+---@param msg string
+local function log_err(msg)
+  vim.notify(msg, vim.log.levels.ERROR, { title = "easy-commands.nvim" })
+end
+
+-- Helper function to check if a table is array-like
+local function is_array_like(tbl)
+  local i = 1
+  for _ in pairs(tbl) do
+    if not tbl[i] then
+      return false
+    end
+    i = i + 1
+  end
+  return true
+end
+
+-- Function to merge two array-like tables
+local function merge_array_like_tables(tbl1, tbl2)
+  local merged = {}
+  local seen = {}
+
+  -- Add elements from tbl1
+  for _, v in ipairs(tbl1) do
+    merged[#merged + 1] = v
+    seen[v] = true
+  end
+
+  -- Add elements from tbl2 that are not already in merged
+  for _, v in ipairs(tbl2) do
+    if not seen[v] then
+      merged[#merged + 1] = v
+      seen[v] = true
+    end
+  end
+
+  return merged
+end
+
+---Merge two table
+---if the field type is string or number or other primary type, use value in tbl1
+---if the field type is array like table, merge and unique the elements,
+---if the field type is table, recursively call mergeTables function
+---
+---	local tbl1 = {
+---		name = "John",
+---		age = 30,
+---		hobbies = { "reading", "swimming" },
+---		details = {
+---			city = "New York",
+---			country = "USA",
+---		},
+---	}
+---
+---	local tbl2 = {
+---		age = 31,
+---		hobbies = { "running", "swimming" },
+---		details = {
+---			city = "San Francisco",
+---			state = "California",
+---		},
+---	}
+---
+---	local expected = {
+---		name = "John",
+---		age = 30,
+---		hobbies = { "reading", "swimming", "running" },
+---		details = {
+---			city = "New York",
+---			country = "USA",
+---			state = "California",
+---		},
+---	}
+---
+---@param tbl1 table
+---@param tbl2 table
+---@return table
+local function merge_tables(tbl1, tbl2)
```

Comment:
Doesn't [`vim.tbl_deep_extend`](https://neovim.io/doc/user/lua.html#vim.tbl_deep_extend%28%29) do the same?

Reply:
if the table has a array-like field, will it merge the array's element to form a new array or just use one array value from one table?
I'll take a look